var/spack/repos/builtin/packages/py-gql/package.py

```diff
@@ -14,10 +14,17 @@ class PyGql(PythonPackage):
     url      = "https://github.com/graphql-python/gql/archive/v3.0.0a1.tar.gz"
 
     version('3.0.0a1', sha256='3254a6010464932e3700a8d225cf6e40a6983aaf5f279615504c8196a374daf9')
+    version('0.4.0', sha256='bd27c44b15b529fecdd80b07291dfd40264a2af937a75b41ff2be49a730e41d8',
+            deprecated=True)
```

Comment:
Why add a deprecated version? If it's needed by some other package, is that package also deprecated then?

Reply:
#23768 (py-sierrapy) requires this exact version else it will not work and that is with the latest version of py-sierrapy.

Reply:
Should we remove `deprecated=True` then?

Reply:
That would make sense since it's needed for sierrapy but I wasn't sure what the Spack convention was for this. I can remove it just let me know what is needed.

Reply:
The convention is to only add `deprecated=True` for things like security flaws or versions we want to remove in future releases, see https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions